Source/Particles/Collision/BinaryCollision/ParticleCreationFunc.H

```diff
@@ -204,14 +204,21 @@ public:
 
                 // If the colliding particle weight decreases to zero, remove particle by
                 // setting its id to -1
-                constexpr amrex::Long minus_one_long = -1;
                 if (w1[p_pair_indices_1[i]] <= amrex::ParticleReal(0.))
                 {
-                    particle_ptr_1[p_pair_indices_1[i]].atomicSetID(minus_one_long);
+                    //particle_ptr_1[p_pair_indices_1[i]].atomicSetID(minus_one_long); // TODO
+                    // or?
+                    //soa_1.m_idata[PIdxInt::id] = -1;
+                    // or?
+                    particle_ptr_1->id() = -1;
```

Comment:
Needs to be at least atomic (why?).
Does it matter if the ID is negative or exactly -1?

cc @NeilZaim

Reply:
Hi @ax3l 

1.  Yes this is to prevent a race condition. This is within a loop on pairs of colliding particles. A single particle can be in multiple pairs so potentially multiple GPU threads could try to set the ID of a single particle to -1 at the same time.
2. Yes, I also think that any negative ID will do here (@atmyers can probably confirm). The id is set to -1 solely for the particle to be removed during redistribute.

Reply:
It does not need to be `-1`, any negative value will cause the redistribute in AMReX to remove that particle. 

Question - what happens if we just let multiple threads set the id to negative 1 simultaneously? It might show up in some profiling software that helps detect race conditions, but is it actually dangerous?

Reply:
I thought these race conditions were always undefined behaviour, even if multiple threads are writing the same value to the variable. If it's not UB, then it's ok to remove the atomic.

Reply:
Multiple accesses to a location, at least one of which is a write, are a data race, resulting in UB. You can specify relaxed atomic stores, but they still must be atomic.

Reply:
Excellent, that sorts it. We will do an atomic to the `id` and keep `cpu` untouched.